testing/test_lightning_components.py

```diff
@@ -0,0 +1,226 @@
+import os
+import yaml
+import torch
+import math
+import pytest
+from pathlib import Path
+from GANDLF.models.lightning_module import GandlfLightningModule
+from GANDLF.losses.loss_calculators import (
+    LossCalculatorFactory,
+    LossCalculatorSimple,
+    LossCalculatorStdnet,
+    AbstractLossCalculator,
+    LossCalculatorDeepSupervision,
+)
+from GANDLF.metrics.metric_calculators import (
+    MetricCalculatorFactory,
+    MetricCalculatorSimple,
+    MetricCalculatorStdnet,
+    MetricCalculatorDeepSupervision,
+    AbstractMetricCalculator,
+)
+from GANDLF.utils.pred_target_processors import (
+    PredictionTargetProcessorFactory,
+    AbstractPredictionTargetProcessor,
+    IdentityPredictionTargetProcessor,
+    DeepSupervisionPredictionTargetProcessor,
+)
+from GANDLF.parseConfig import parseConfig
+from GANDLF.utils.write_parse import parseTrainingCSV
+from GANDLF.utils import populate_header_in_parameters
+
+testingDir = Path(__file__).parent.absolute().__str__()
+
+
+def add_mock_config_params(config):
+    config["penalty_weights"] = [0.5, 0.25, 0.175, 0.075]
+    config["model"]["class_list"] = [0, 1, 2, 3]
```

Comment:
This can be changed based on the support for multi-layer deep supervision.

Reply:
Could you explain/point me towards the part of docs saying more? Not sure if I fully get that and what should be specified there.